infra/aws/terraform/modules/eks-prow-iam/policies.tf

```diff
@@ -0,0 +1,235 @@
+      "iam:GetOpenIDConnectProvider",
+      "iam:GetPolicy",
+      "iam:GetPolicyVersion",
+      "iam:GetRole",
+      "iam:GetRolePolicy",
+      "iam:GetUser",
+      "iam:ListAttachedRolePolicies",
+      "iam:ListRolePolicies",
+      "iam:PassRole",
+      "iam:PutRolePolicy",
+      "iam:TagOpenIDConnectProvider",
+      "iam:TagPolicy",
+      "iam:TagRole",
+      "iam:UpdateOpenIDConnectProviderThumbprint",
+      "kms:CreateAlias",
+      "kms:CreateGrant",
+      "kms:CreateKey",
+      "kms:DescribeKey",
+      "kms:EnableKeyRotation",
+      "kms:GetKeyPolicy",
+      "kms:GetKeyRotationStatus",
+      "kms:ListAliases",
+      "kms:ListResourceTags",
+      "kms:PutKeyPolicy",
+      "kms:TagResource",
+      "logs:CreateLogGroup",
+      "logs:DescribeLogGroups",
+      "logs:ListTagsLogGroup",
+      "logs:PutRetentionPolicy",
+      "s3:GetObject",
+      "s3:ListBucket",
+      "s3:PutObject"
+    ]
+  }
+}
+
+data "aws_iam_policy_document" "eks_destroy" {
+  statement {
+    sid       = ""
+    effect    = "Allow"
+    resources = ["*"]
+
+    actions = [
+      "ec2:DeleteEgressOnlyInternetGateway",
+      "ec2:DeleteInternetGateway",
+      "ec2:DeleteLaunchTemplate",
+      "ec2:DeleteNatGateway",
+      "ec2:DeleteRoute",
+      "ec2:DeleteRouteTable",
+      "ec2:DeleteSecurityGroup",
+      "ec2:DeleteSubnet",
+      "ec2:DeleteVpc",
+      "ec2:DescribeInternetGateways",
+      "ec2:DescribeNatGateways",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DescribeRouteTables",
+      "ec2:DescribeSecurityGroups",
+      "ec2:DescribeVpcs",
+      "ec2:DetachInternetGateway",
+      "ec2:DisassociateAddress",
+      "ec2:DisassociateRouteTable",
+      "ec2:DisassociateVpcCidrBlock",
+      "ec2:ReleaseAddress",
+      "ec2:RevokeSecurityGroupEgress",
+      "ec2:RevokeSecurityGroupIngress",
+      "eks:DeleteAddon",
+      "eks:DeleteCluster",
+      "eks:DeleteNodegroup",
+      "eks:DescribeAddon",
+      "eks:DescribeCluster",
+      "eks:DescribeNodegroup",
+      "iam:DeleteOpenIDConnectProvider",
+      "iam:DeletePolicy",
+      "iam:DeleteRole",
+      "iam:DeleteRolePolicy",
+      "iam:DetachRolePolicy",
+      "iam:ListAttachedRolePolicies",
+      "iam:ListInstanceProfilesForRole",
+      "iam:ListPolicyVersions",
+      "iam:ListRolePolicies",
+      "kms:DeleteAlias",
+      "kms:DescribeKey",
+      "kms:ScheduleKeyDeletion",
+      "logs:DeleteLogGroup",
+      "s3:PutObject"
+    ]
+  }
+}
+
+resource "aws_iam_policy" "eks_plan" {
+  name   = "TFProwClusterViewer"
+  path   = "/"
+  policy = data.aws_iam_policy_document.eks_plan.json
+  tags   = var.tags
+}
+
+resource "aws_iam_policy" "eks_apply" {
+  name   = "TFProwClusterApply"
+  path   = "/"
+  policy = data.aws_iam_policy_document.eks_apply.json
+  tags   = var.tags
+}
+
+resource "aws_iam_policy" "eks_destroy" {
+  name   = "TFProwClusterDestroy"
+  path   = "/"
+  policy = data.aws_iam_policy_document.eks_destroy.json
+  tags   = var.tags
+}
```

Comment:
Should we rename those policies to be in line with role names, e.g. `EKSClusterViewer`, `EKSClusterApplier`, EKSClusterDestoryer`? Is that too complicated at this point (if it makes sense at all)?